### PR TITLE
Update the OSF Setup guide

### PR DESCRIPTION
**Pull request by cameronblandford**
Added information where it was missing, normalized some code styling, and fixed a typo.

### CODE REVIEW

**Comment by cameronblandford** [comment]
To Whom It May Concern (probably Steve),

Thoughts?

File: docs/osf/setup.rst [hunk 1]
```diff
@@ -326,7 +329,15 @@ Installing libxml2 and libxslt
 Install node packages with ``npm``
 ----------------------------------
 
-``npm`` is used to install required Node.JS packages.
+``npm`` (node package manager) is used to install required Node.JS packages. To install ``npm``, you have to install ``node.js``, which includes ``npm``. `npmjs.org <http://blog.npmjs.org/post/85484771375/how-to-install-npm>`_ recommends doing this through the `node.js installer <https://nodejs.org/en/>`_. Be sure to check for upgrades after installation by running: 
+
+    ::
+
+        sudo npm install npm -g
```
Comment (sloria) [inline docs/osf/setup.rst]: Is this the recommended way to update npm? `sudo` is generally discouraged when using Homebrew.

**Comment by sloria** [comment]
Thanks for the update. Looks good; only concern is https://github.com/CenterForOpenScience/COSDev/pull/41/files#r74644349

**Comment by felliott** [comment]
@cameronblandford++!  (documentation iz life)

**Pull request merged**